### PR TITLE
Optimized single played conditions

### PR DESCRIPTION
**Pull request by dvdvideo1234**

### CODE REVIEW

**Comment by dvdvideo1234** [comment]
@thegrb93  Is it worth it to create and destroy `EffectData()` at every shot ??

**Comment by thegrb93** [comment]
Sure

**Comment by thegrb93** [comment]
Better yet to initialize it in the Initialize function. Will eliminate the conditional there.

**Comment by thegrb93** [comment]
Same goes for spreadvector.

**Comment by dvdvideo1234** [comment]
@thegrb93 

The vector is created, set up by `self:SetSpread(<number> spread)`. If no vector is present, crates one with the `spread` else updates the already created vector and you just call `SetSpread`.

**Comment by thegrb93** [comment]
If you create them in Initialize, the conditional isn't needed. It should be in Initialize, not Setup.

**Comment by dvdvideo1234** [comment]
@thegrb93  Thanks, done ;)

**Comment by dvdvideo1234** [comment]
@thegrb93 

Is it worth it to set the sound to `nil` when it becomes empty string, so that `EmitSound` will never get called in such case ?

File: lua/entities/gmod_wire_turret.lua [hunk 1]
```diff
@@ -96,47 +96,50 @@ local ValidTracers = {
 	[""]                      = true
 }
 
-function ENT:SetSound(sound)
+function ENT:SetSound( sound )
+	local sound = tostring( sound or "" )
+	sound = string.Trim( sound ) -- The string cannot have whitespace
 	local check = string.find( sound, "[\"?]" ) -- Preventing client crashes
-	self.sound = check and "" or sound
+	sound = check and "" or sound
```
Comment (thegrb93) [inline lua/entities/gmod_wire_turret.lua]: self.sound = check~=nil and sound~="" and sound or nil

Comment (dvdvideo1234) [reply lua/entities/gmod_wire_turret.lua]: https://github.com/wiremod/wire/pull/2075/commits/4b364bbd0bc9870d86ab82368f80b4e3cef058aa

**Comment by thegrb93** [comment]
I'm wondering if that `string.find( sound, "[\"?]" )` should be `string.find( sound, "[\"%?]" )`

**Comment by dvdvideo1234** [comment]
It was '["?]' before for containing " or EOL. So I used double quotes to escape the ".

**Comment by thegrb93** [comment]
Yeah looks like either works.

**Pull request merged**